gflownet/utils/batch.py

```diff
@@ -1062,6 +1066,48 @@ def is_valid(self) -> bool:
             return False
         return True
 
+    def traj_indices_are_consecutive(self) -> bool:
+        """
+        Returns True if the trajectory indices start from 0 and are consecutive; False
+        otherwise.
+        """
+        return set(self.trajectories) == set(np.arange(self.get_n_trajectories()))
+
+    def make_indices_consecutive(self):
```

Comment:
This is something for which a small unit test (with multiple expected cases) would be great.

Reply:
Not so small...

Reply:
[There](https://github.com/alexhernandezgarcia/gflownet/pull/167/commits/28fa35c2c789253edc43e674144c5ca5dfabfd1c) you go.

Reply:
And note that I have reimplemented things:
- https://github.com/alexhernandezgarcia/gflownet/pull/167/commits/b28ded1da22ddec29db00fba8e116feecae01100
- https://github.com/alexhernandezgarcia/gflownet/pull/167/commits/27a5893b631729b6aa5ef2deed3d7d2dab7d5d73

Reply:
And finally, note that `make_indices_consecutive()` is now not needed after all... (https://github.com/alexhernandezgarcia/gflownet/pull/167/commits/627ee04aebc6eda09e5ae45dd9299a97c691036f)